src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+
+    public Figure getRandomFigure() {
+        int randomType = RANDOM.nextInt(5);
+        String randomColor = new ColorSupplier().getRandomColor();
+        switch (randomType) {
+            case 0:
+                return new Square(RANDOM.nextDouble() * 10, randomColor);
+            case 1:
+                return new Rectangle(RANDOM.nextDouble() * 10,
+                        RANDOM.nextDouble() * 10, randomColor);
+            case 2:
+                return new RightTriangle(RANDOM.nextDouble() * 10 + 1,
+                        RANDOM.nextDouble() * 10 + 1, randomColor);
+            case 3:
+                return new Circle(RANDOM.nextDouble() * 10, randomColor);
+            default:;
+                return new IsoscelesTrapezoid(
+                        RANDOM.nextDouble() * 10, RANDOM.nextDouble() * 10,
+                        RANDOM.nextDouble() * 10, randomColor);
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, "white");
+    }

```

Comment:
1. Magic number should be constants with proper naming
2. Services our class rely on are usually private final as they are mutable and cannot be fully constants
3. Initialization of service with new in method will cause creation of new object each time method is called so should be moved to class level to avoid this
4. As you require colors to be available here and in ColorSupplier it's better to create enum with colors in order not to hardcode values in different places

```suggestion
public class FigureSupplier {
    private static final int SIDE_LIMIT = 10;
    private static final String DEFAULT_COLOR = Color.WHITE.name();
    private static final int DEFAULT_RADIUS = 10;
    
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();

    public Figure getRandomFigure() {
        int randomType = random.nextInt(Color.values.length);
        String randomColor = colorSupplier.getRandomColor();
        switch (randomType) {
            case 0:
                return new Square(getRandomSide(), randomColor);
            case 1:
                return new Rectangle(getRandomSide(),
                        getRandomSide(), randomColor);
            case 2:
                return new RightTriangle(getRandomSide(),
                        getRandomSide(), randomColor);
            case 3:
                return new Circle(getRandomSide(), randomColor);
            default:;
                return new IsoscelesTrapezoid(
                        getRandomSide(), getRandomSide(),
                        getRandomSide(), randomColor);
        }
    }

    public Figure getDefaultFigure() {
        return new Circle(DEFAULT_RADIUS, DEFAULT_COLOR);
    }
    
    private double getRandomSide() {
        return random.nextDouble() * SIDE_LIMIT;
    }
```